app/helpers/accounts_helper.rb

```diff
@@ -0,0 +1,23 @@
+module AccountsHelper
+  def login_field(form)
+    form.text_field :login, size: 25, required: true
+  end
+
+  def email_field(form)
+    form.text_field :mail, required: true
+  end
+
+  ##
+  # Hide the email field in the registration form if the `email_login` setting
+  # is active. However, if an auth source is present, do show it independently from
+  # the `email_login` setting as we can't say if the auth source's login is the email address.
+  def registration_show_email?
+    !Setting.email_login? || @user.auth_source_id.present?
+  end
+
+  def registration_footer
+    footer = Setting.registration_footer[I18n.locale.to_s].presence
+
+    RedCloth3.new(footer).to_html if footer
```

Comment:
`OpenProject::TextFormatting.format_text` is the OP spiced wrapper around RedCloth3 and should be used throughout the application.